tools/downloader/networks_compiler.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import sys
+import common
+from pathlib import Path
+
+def compile(reporter, compiler_path, model, model_precision, args, output_dir):
+    (output_dir / model.subdirectory).mkdir(parents=True, exist_ok=True)
+
+    compile_cmd = [str(compiler_path),
+        '-m={}'.format(args.model_dir / model.subdirectory / model_precision / (model.name + '.xml')),
+        '-d={}'.format(args.target_device),
+        '-ip={}'.format('U8' if args.input_precision is None else args.input_precision),
+        '-op={}'.format('FP32' if args.output_precision is None else args.output_precision),
```

Comment:
Why do we specify our own defaults? It looks like `compile_tool` already has its own defaults for these options, are they unsuitable?

Reply:
With default compiled blobs for MYRIAD demos doesn't work due to inputs and ouptuts precisions are FP16 and in IE cast from FP16 to FP32 is absent,

Reply:
Okay; in that case, I have other questions:

* Why are you using U8 as the default input precision? Presumably, there's no cast from FP32 to U8 either; moreover, this will only work for models with inputs that fit into the U8 datatype (i.e. images).
* Why are these defaults customizable, and why are there specific options for it? If there is utility in letting the user override the input/output precisions, then maybe a generic extra option mechanism (like `--add_mo_arg` in Model Converter) would be more flexible?

Reply:
> Why are you using U8 as the default input precision? 

Blobs support added only in demos where network's input is U8.

> Why are these defaults customizable, and why are there specific options for it?

I think I can change it to an extra option like` --add_mo_arg`

Reply:
Changed